### DISCUSSION TITLE
Best practice for tackling different python versions?

### DISCUSSION THREAD

**Discussion by AnatolyBuga** [PyO3/maturin#1411] [Q&A] [answered]
Hello Team and thanks for this amazing tool,

I have uploaded a test packange `https://test.pypi.org/project/ultibi/` to `testpypi`. I did so with -i pointing to Python 3.9 interpereter. _I assumed that would mean that anyone with Python >= 3.9 (for example 3.11) would be able to use it_. I think this is how polars did it [here](https://github.com/pola-rs/polars/blob/master/.github/workflows/create-python-release.yaml).  Note they always use: `python-version: '3.7' ` However, it doesn't seem to work:
`pip install -i https://test.pypi.org/simple/ ultibi==0.1.2` with a python 3.9 throws an error: `No matching distribution found for maturin<0.15,>=0.14`. 

**How do I make my package work with every Python>=3.9?**

Thanks!

**Comment by AnatolyBuga** [comment, marked as answer]
Just noticed this [question](https://github.com/PyO3/maturin/discussions/1125). I can confirm that adding `"abi3-py39"` to pyo3 features solved my problem and made my wheel work with Python 3.11.

**Reply by messense** [reply]
See also https://pyo3.rs/v0.17.3/building_and_distribution#py_limited_apiabi3
